backend/infrahub/core/schema/schema_branch.py

```diff
@@ -995,6 +998,50 @@ def validate_kinds(self) -> None:
                         f"{node.kind}: Relationship {rel.name!r} is referring an invalid peer {rel.peer!r}"
                     ) from None
 
+    def validate_attribute_parameters(self) -> None:
+        for name in self.generics.keys():
+            generic_schema = self.get_generic(name=name, duplicate=False)
+            for attribute in generic_schema.attributes:
+                if (
+                    attribute.kind == "NumberPool"
+                    and isinstance(attribute.parameters, NumberPoolParameters)
+                    and not attribute.parameters.number_pool_id
+                ):
+                    attribute.parameters.number_pool_id = str(uuid4())
```

Comment:
Could `parameters.number_pool_id` somehow have its default value set to `str(uuid4())`? This would avoid this code and mutating the object after its creation.

Reply:
I don't think so as we don't want to set the value in the model if it has already been assigned in the database.